johko/computer-vision-course
# chore: add linters

Fixes https://github.com/johko/computer-vision-course/issues/16

This PR integrates `black` and `ruff` as pre-commit hooks to improve our code quality and consistency. With these hooks in place, every code change will be automatically formatted according to the standards set by `black` and validated by `ruff` before commits.

## Changes
1. Added `black` as a pre-commit hook.
   - Ensures consistent code formatting across the codebase.
   - Adheres to the PEP 8 styling guide with minor modifications for readability.
   
2. Added `ruff` as a pre-commit hook.
   - Provides linting checks and highlights potential issues in the code.
   - Helps maintain clean and maintainable code practices.

## Benefits
- **Consistency**: With `black`, we can ensure that our Python codebase looks the same regardless of the author.
- **Quality**: `ruff` will help catch potential errors or anti-patterns early, ensuring high-quality code.
- **Productivity**: Developers can focus more on logic and less on formatting or minor linting issues. Any issues will be caught early, reducing back-and-forth during code reviews.

## How to Test
1. Check out this branch.
2. Install pre-commit: `pip install pre-commit`.
3. Set up the git hook scripts: `pre-commit install`.
4. Make changes to any `.py` or `.ipynb` file.
5. Try committing your changes. The pre-commit hooks should run `black` and `ruff`.

Feedback and suggestions are welcome!

Thank you.
